react-first-app/src/app/components/App/Game/index.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react';
+
+import logo from '../../../../logo.svg';
+
+import style from './styles.css';
+
+import Board from '../Board';
+
+class Game extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      history: [
+        {
+          squares: Array(9).fill(null)
+        }
+      ],
+      stepNumber: 0,
+      xIsNext: true
+    };
+  }
+
+  handleClick(i) {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
```

Comment:
if you want to make a copy, you could just `const squares = [...current.squares]`